src/hyperion/external_interaction/ispyb/store_in_ispyb.py

```diff
@@ -311,24 +327,31 @@ def __init__(
         self,
         ispyb_config: str,
         experiment_type: str,
-        parameters: GridscanInternalParameters = None,
+        parameters: GridscanInternalParameters | None = None,
```

Comment:
Should: When do we ever pass `None` into here in production? Do we need to allow it?

Reply:
good spot, I cleaned up a bunch of other ones in here too